### DISCUSSION TITLE
GARO Entity Pro

### DISCUSSION THREAD

**Discussion by Jerker74** [lbbrhzn/ocpp#1037] [Show and tell]
Two months ago I got my GARO Entity Pro installed in the new house. Yesterday I tried to connect it to this OCPP integration. I still have some issues to sort out but I working on it...

I might ask some people for help in the threads, I'm an experienced developer but new to this context.

![Screenshot_20240108-074904](https://github.com/lbbrhzn/ocpp/assets/91065694/9eae6ebc-0c70-40df-bac0-6c1cbb195288)

**Comment by munterkalmsteiner** [comment]
Got it also installed just before Christmas. Fixed now an [issue with the load balancer](https://github.com/lbbrhzn/ocpp/issues/1043) and found out how to [charge with one phase](https://github.com/lbbrhzn/ocpp/issues/1035) if needed.

**Comment by pieth861** [comment]
I have a Garo Enitiy Pro an managed to extract data thru the local API. But have not managed to set up the inte gration with HA. I did not manage to figure out how to set up the Garo Entity Pro properly.

@Jerker74, I would appreciate if you can give some description on the setup on the Garo Enitiy Pro for connection with OCPP to HA?

**Comment by munterkalmsteiner** [comment]
@pieth861 do you have also a load balancer installed? If yes, check out this [issue](https://github.com/lbbrhzn/ocpp/issues/1043). If not, how far did you get with your setup?

**Reply by pieth861** [reply]
Hi @munterkalmsteiner , I don't get a connection at all between HA and the Garo Entity PRO. I have restarted both HA and Garo and put the IP of the HA  in at "GaroRemoteCsmsUrl". 
I have a Balance Basic installed as the master in the system. 

I will check the issue you referred to. Thanks for fast response.

**Reply by Jerker74** [reply]
Hi pieth861,

Did you sort it out? I'm not the right person to ask unfortunately. I have not managed to get it working according to Michaels instruction or any other way. But if you do I'm really interested in how you did.

**Reply by Jerker74** [reply]
As I wrote in the other thread I have now managed to connect. Still not able to control it or get it to charge.

**Reply by pieth861** [reply]
Hi @Jerker74, then we ar at the same state. I also got it to connect but I'm not able to control the charger. At the same time the charger goes offline in the Garo app. I don't have time right now to continue the struggle at the moment. But, please enlighten us with any progress. 😀

**Reply by Jerker74** [reply]
Hi, 
I have been in contact with a person working with integrating GARO/OCPP systems and it's not good news. According to him the OCPP server needs to be able to talk to both the CS and the DLM. So... as @munterkalmsteiner suggested the OCPP server integration needs to be upgraded to handle multiple devices. Otherwise the Garo Entity Pro + Load balancer will not be able to integrate into Home Assistant.

Another option could be to get Garo to help out with a proxy that is able to manage this communication setup.

**Reply by munterkalmsteiner** [reply]
I'm currently working on a fix, based on the work by @lbbrhzn which was 99% ready. I just smoothed out a few bumps. Here is my [branch](https://github.com/munterkalmsteiner/ocpp/tree/new_option_flows_updated).

**Reply by Jerker74** [reply]
Hi @munterkalmsteiner,
I haven't been around for a while and I saw now there are updates in the master. Just curious, how is your branch working?

**Reply by munterkalmsteiner** [reply]
Unfortunately I'm too busy with my day job at the moment. I did not forget it though.
